src/pgo/formatters/IssueFormattingVisitor.java

```diff
@@ -82,10 +89,15 @@ public Void visit(IOErrorIssue ioErrorIssue) throws IOException {
 	}
 
 	@Override
-	public Void visit(TLAParserIssue tlaParserIssue) throws IOException {
-		out.write("could not parse TLA: ");
+	public Void visit(ParsingIssue parsingIssue) throws IOException {
+		Map.Entry<SourceLocation, Set<ParseFailure>> lastEntry = parsingIssue.getError().lastEntry();
+		out.write("could not parse "+parsingIssue.getLanguage()+" at "+lastEntry.getKey()+": ");
 		try(IndentingWriter.Indent ignored = out.indent()){
-			out.write(tlaParserIssue.getError().toString());
+			Set<ParseFailure> lastFailures = lastEntry.getValue();
```

Comment:
Why choose a set? This places the burden of implementing sensible logic for equality and hashing on the next guy.

Reply:
The set was chosen since it is common for there to be many instances of the same parse failure in the same place - it forces deduplication of the failures on store. I do know it's not very elegant (unordered, for example), but I'm not sure about what it means to place a burden on the next guy... I've already implemented hashing and equality for all failure types?